gateway/src/cron.rs

```diff
@@ -189,13 +201,13 @@ impl CronSubmission {
         // consider an example that consensus will never be reached:
         //
         // -------- | -------------------------|--------------- | ------------- |
-        //     MOST_VOTED                 THRESHOLD     TOTAL_SUBMISSIONS  TOTAL_VALIDATORS
+        //     MOST_VOTED                 THRESHOLD     TOTAL_SUBMISSIONS  TOTAL_WEIGHT
         //
-        // we see MOST_VOTED is smaller than THRESHOLD, TOTAL_SUBMISSIONS and TOTAL_VALIDATORS, if
-        // the potential extra votes any vote can obtain, i.e. TOTAL_VALIDATORS - TOTAL_SUBMISSIONS,
+        // we see MOST_VOTED is smaller than THRESHOLD, TOTAL_SUBMISSIONS and TOTAL_WEIGHT, if
+        // the potential extra votes any vote can obtain, i.e. TOTAL_WEIGHT - TOTAL_SUBMISSIONS,
         // is smaller than or equal to the potential extra vote the most voted can obtain, i.e.
         // THRESHOLD - MOST_VOTED, then consensus will never be reached, no point voting, just abort.
-        if threshold - most_voted_count >= total_validators - self.total_submissions {
+        if threshold - most_voted_weight >= total_weight - self.total_submission_weight.clone() {
```

Comment:
Like above, I think this works if I am not wrong.
```suggestion
        if threshold - most_voted_weight >= total_weight - &self.total_submission_weight {
```